Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -258,7 +258,9 @@ open class NavigationMapView: UIView {
                 }
                 
                 if showsRestrictedAreasOnRoute {
-                    parentLayerIdentifier = addRouteRestrictedAreaLayer(route, below: parentLayerIdentifier)
+                    parentLayerIdentifier = addRouteRestrictedAreaLayer(route,
+                                                                        below: parentLayerIdentifier,
+                                                                        reuseExistingLayer: true)
```

Comment:
An alternative to re-using existing layers is to forcefully remove existing `lineLayer`s for both `routes` and `alternativeRoutes` at the beginning of `applyRoutesDisplay`. But this seems to create an undesired overhead since during setting new routes or alternative routes view will remove and re-add all of the layers.
Can someone see a case when we would actually still want to remove existing layer even though it's `route` was not changed?